lib/test_data_populator.rb

```diff
@@ -0,0 +1,99 @@
+# test_data_populator.rb
+#
+# creates model instances as needed for tests without having to use seeds.rb
+# can add example files to an already existing study in the test environment.  this differs from FactoryBot in that it is
+# intended for regular, non-detached studies as it will push physical files to workspace buckets.  this also differs
+# from SyntheticStudyPopulator in that it does not parse files
+# can also create the 3 default SearchFacet instances for species, disease, and organism_age commonly used in tests
+class TestDataPopulator
+
+  # normal example test files
+  EXAMPLE_FILES = {
+    expression: {
+      name: 'expression_matrix_example.txt',
+      path: 'test/test_data/expression_matrix_example.txt',
+      file_type: 'Expression Matrix'
+    },
+    metadata: {
+      name: 'metadata_example.txt',
+      path: 'test/test_data/alexandria_convention/metadata.v2-0-0.txt',
+      file_type: 'Metadata',
+      use_metadata_convention: true
+    },
+    cluster: {
+      name: 'cluster_example.txt',
+      path: 'test/test_data/cluster_example.txt',
+      file_type: 'Cluster'
+    }
+  }.with_indifferent_access.freeze
+
+  FILE_TYPES = EXAMPLE_FILES.keys.freeze
+
+  # add example files to an existing, non-detached study and push to workspace bucket
+  def self.add_files_to_study(study, file_types: FILE_TYPES)
```

Comment:
this is nice functionality. I think we can include this capability in our study factory.  If we make a new 'predefined_file_types' transient argument to the study factory, and if present, add the files with the code below in the existing `after(:create)` method.  That keeps more of our study-testing code in the same place.

Reply:
(and you may want to still keep this function accessible as an independent method for the cases where someone wants to create a study with no files, test something, add files, and then test something else

Reply:
Yeah let me kick the tires on that - it's a fair point.

Reply:
Actually, I think I'm going to leave the code in place and reference it from the `Study` factory.  This way the same code is exercised/tested, and it can be used both via tests or manually.